src/main/java/org/embulk/output/elasticsearch/ElasticsearchHttpClient.java

```diff
@@ -0,0 +1,342 @@
+        }
+        for (JsonNode index : response) {
+            if (index.has("aliases") && index.get("aliases").has(aliasName)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public void assignAlias(String indexName, String aliasName, PluginTask task, Jetty92RetryHelper retryHelper)
+    {
+        try {
+            if (isExistsIndex(indexName, task, retryHelper)) {
+                if (isExistsAlias(aliasName, task, retryHelper)) {
+                    // curl -XPUT http://localhost:9200/_alias -d\
+                    // "actions" : [
+                    //   {"remove" : {"alias" : "{alias}", "index" : "{index_old}"}},
+                    //   {"add" : {"alias": "{alias}", "index": "{index_new}"}}
+                    // ]
+                    // Success: {"acknowledged":true}
+                    List<String> oldIndices = getIndexByAlias(aliasName, task, retryHelper);
+
+                    Map<String, String> newAlias = new HashMap<>();
+                    newAlias.put("alias", aliasName);
+                    newAlias.put("index", indexName);
+                    Map<String, Map> add = new HashMap<>();
+                    add.put("add", newAlias);
+
+                    Map<String, String> oldAlias = new HashMap<>();
+                    // TODO multiple alias?
+                    for (String oldIndex : oldIndices) {
+                        oldAlias.put("alias", aliasName);
+                        oldAlias.put("index", oldIndex);
+                    }
+                    Map<String, Map> remove = new HashMap<>();
+                    remove.put("remove", oldAlias);
+
+                    List<Map<String, Map>> actions = new ArrayList<>();
+                    actions.add(remove);
+                    actions.add(add);
+                    Map<String, List> rootTree = new HashMap<>();
+                    rootTree.put("actions", actions);
+
+                    String content = jsonMapper.writeValueAsString(rootTree);
+                    sendRequest("/_aliases", HttpMethod.POST, task, retryHelper, content);
+                    log.info("Re-Assigned alias [{}] to index[{}]", aliasName, indexName);
+                }
+                else {
+                    // curl -XPUT http://localhost:9200/{index}/_alias/{alias}
+                    // Success: {"acknowledged":true}
+                    String path = String.format("/%s/_alias/%s", indexName, aliasName);
+                    sendRequest(path, HttpMethod.PUT, task, retryHelper);
+                    log.info("Assigned alias [{}] to Index [{}]", aliasName, indexName);
+                }
+            }
+        }
+        catch (JsonProcessingException ex) {
+            throw new ConfigException(String.format("Failed to assign alias[%s] to index[%s]", aliasName, indexName));
+        }
+    }
+
+    public void reAssignAlias(String aliasName, String newIndexName, PluginTask task, Jetty92RetryHelper retryHelper)
```

Comment:
We may want to have a comment (Javadoc) which describes this method should be called just once, in `transaction`, not in each Task.